joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```
@@ -64,85 +64,91 @@ def init_collada(self, robot):
                     maxval = float(limit.getElementsByTagName('max')[0].childNodes[0].nodeValue)
-                if minval == maxval:  # this is a fixed joint
-                    continue
+            if minval == maxval:  # this is a fixed joint
+                continue
 
-                self.joint_list.append(name)
-                minval *= math.pi/180.0
-                maxval *= math.pi/180.0
-                self.free_joints[name] = self._init_joint(minval, maxval, 0.0)
+            self.joint_list.append(name)
+            minval *= math.pi/180.0
+            maxval *= math.pi/180.0
+            self.free_joints[name] = self._init_joint(minval, maxval, 0.0)
 
     def init_urdf(self, robot):
         robot = robot.getElementsByTagName('robot')[0]
         # Find all non-fixed joints
         for child in robot.childNodes:
             if child.nodeType is child.TEXT_NODE:
                 continue
-            if child.localName == 'joint':
-                jtype = child.getAttribute('type')
-                if jtype in ['fixed', 'floating', 'planar']:
-                    continue
-                name = child.getAttribute('name')
-                self.joint_list.append(name)
-                if jtype == 'continuous':
-                    minval = -math.pi
-                    maxval = math.pi
-                else:
-                    try:
-                        limit = child.getElementsByTagName('limit')[0]
+            if child.localName != 'joint':
+                continue
+            jtype = child.getAttribute('type')
+            if jtype in ['fixed', 'floating', 'planar']:
+                continue
+            name = child.getAttribute('name')
+            self.joint_list.append(name)
+            if jtype == 'continuous':
+                minval = -math.pi
+                maxval = math.pi
+            else:
+                try:
+                    limit_list = child.getElementsByTagName('limit')
+                    if limit_list:
```

This is actually subtly changing the semantics here.  Previously, if no limit was specified, then this would probably throw an `IndexError`, which would be captured by the `except` clause below and the joint ignored.  Now, if there is no limit it will silently go on without limits.  I think we should preserve the old behavior here.

---

I was slightly hesitant to make this change just for consistency. I don't recall why I started out to update `init_collada` probably to clear some warning. We could either have an extra variable in `init_urdf` to preserve the old behaviour (recent commit) or revert from all three functions.

---

The `init_collada` case is a little more interesting.  Since it doesn't have the `try/except` block, if it was missing limits it would previously just throw an exception and crash with an `IndexError`.  But I think it would be nice to make it ignore the joint instead, more like what `init_urdf` here does.  So from that perspective maybe we ought to have it print a warning when it is ignoring the joint because of missing limits.

`init_sdf` is a special case because it is brand new.  So we can define whatever semantics we want there.